extra-scripts/comment-deployment-plan-pr.py

```diff
@@ -0,0 +1,112 @@
+"""
+During our CI/CD deployment pipeline, we create a deployment plan that describes
+which hubs will be upgraded as a result of a merge. This script is run in CI/CD
+to download the markdown content of the deployment plan from a GitHub Actions
+workflow artifact and then post it as a comment on the PR that generated the plan.
+"""
+import io
+import os
+import sys
+import zipfile
+
+import requests
+
+# === Setup variables ===#
+# GITHUB_REPOSITORY = ${{ github.repository }} in workflow file
+repo = os.environ.get("GITHUB_REPOSITORY", None)
+if repo is None:
+    raise ValueError("GITHUB_REPOSITORY env var must be set")
+
+# RUN_ID = ${{ github.event.workflow_run.id }} in workflow file
+run_id = os.environ.get("RUN_ID", None)
+if run_id is None:
+    raise ValueError("RUN_ID env var must be set")
+
+# GITHUB_TOKEN = ${{ secrets.GITHUB_TOKEN }} in workflow file
+token = os.environ.get("GITHUB_TOKEN", None)
+if token is None:
+    raise ValueError("GITHUB_TOKEN env var must be set")
+
+api_url = "https://api.github.com"
+
+# Construct headers to send with requests
+headers = {
+    "Accept": "application/vnd.github+json",
+    "Authorization": f"Bearer {token}",
+}
+
+# List all artifacts for the workflow run
+url = "/".join([api_url, "repos", repo, "actions", "runs", run_id, "artifacts"])
+params = {"per_page": 100}
+response = requests.get(url, headers=headers, params=params)
+all_artifacts = response["artifacts"]
+
+# If "Link" is present in the response headers, that means that the results are
+# paginated and we need to loop through them to collect all the results.
+# It is unlikely that we will have more than 100 artifact results for a single
+# worflow ID however.
+while "Link" in response.headers.keys():
+    response = requests.get(response.headers["Link"], headers=headers)
+    all_artifacts.extend(response["artifacts"])
```

Comment:
This pattern won't work - working on a bugfix

Reply:
I believe this is fixed by [`d5d0afa` (#2667)](https://github.com/2i2c-org/infrastructure/pull/2667/commits/d5d0afacb086bc5f8a70c96461bebbd83066917e)